### DISCUSSION TITLE
Which library needed to compile for Wayland? ("Unable to find wayland-scanner and xdg-shell protocol")

### DISCUSSION THREAD

**Discussion by Tahutipai** [brndnmtthws/conky#1629] [Q&A] [answered]
Fresh compile of Conky. Ubuntu 23.04

I have managed to add all needed libaries (so far), but I am stuck on the following message, and desite significant searching, I have not been able to find the correct (apt install...) packages I need to add so I can compile conky.
`Unable to find wayland-scanner and xdg-shell protocol`

I tried adding packages "librust-wayland-scanner-dev" & "wayland-scanner++", which did not resolve the issue.  Can anyone advise me what packages I need to add?

(Also, it would be fantastic if the error messages in `ConkyPlatformChecks.cmake` provided a clue as to what package you likely needed...)

**Comment by Tahutipai** [comment, marked as answer]
Found the answer, for anyone else looking:
`sudo apt get wayland-protocols`

Is there any reason that 'ConkyPlatformChecks.cmake' couldn't be updated to have more helpful error messages - i.e tell the user the name of the package that needs installed?

**Reply by brndnmtthws** [reply]
The package names are distribution dependent, but the answer is now available here for anyone else who has the problem.
